src/ubase/trace.ml

```diff
@@ -15,6 +15,10 @@
     along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *)
 
+(* ignore SIGUSR1 by default, until the logfile is opened, where a signal
+   handler is installed *)
+open Sys;;
+signal sigusr1 Signal_ignore;;
```

Comment:
Why do you need to ignore the signal at all? What happens if you don't?

Reply:
The signal handler is only installed as soon as the logfile is opened. If you don't explicitly ignore the signal, it will call the default handler and terminate the program.

Reply:
Ok, I see. But why don't you install the handler immediately? There is no dependency on the logfile being opened, as far as I can see.